backend/src/index.ts

```diff
@@ -1,16 +1,70 @@
-import express, { Response } from "express";
+import express, { Response, Request } from "express";
 import session from "express-session";
 import { TypedRequest, LoginBody } from "./requestTypes";
+import { SessionStorage, User, LoginErrors } from "./interfaces";
 import bcrypt from "bcrypt";
-import { collection, addDoc, getDocs, where, query, doc, deleteDoc } from "firebase/firestore";
+import {
+  collection,
+  addDoc,
+  getDocs,
+  where,
+  query,
+  deleteDoc,
+  getDoc,
+  doc,
+  Timestamp,
+} from "firebase/firestore";
 import { db } from "./firebase";
 import cors from "cors";
 import crypto from "crypto";
 
-const EXPRESS_PORT = 3000;
+const session_auth = async (sessionId: string) => {
+  const sessions = collection(db, "sessions");
+  const sessionData = query(sessions, where("sessionId", "==", sessionId));
+  const session = await getDocs(sessionData);
+
+  if (session.empty) {
+    return false;
+  }
+
+  // if current time is greater than expiration date, return false and
+  // removes it from the database
+  if (new Date() > session.docs[0].data().expirationDate.toDate()) {
+    await session_remove(sessionId);
+    return false;
+  }
+  return true;
+};
+
+// Clears all expired sessions in the db
+// Idea, periodically call this to stay efficient on document reads?
+const clean_sessions = async () => {
+  const sessions = collection(db, "sessions");
+  const allSessions = await getDocs(sessions);
+  allSessions.forEach(async (sessionDoc) => {
```

Comment:
why doesn't this use session_remove?

Reply:
also doesn't this and session_auth do the same thing

Reply:
i was just thinking about situations where the session db will get bigger and bigger but yea we can deal with that later I'll delete that